auth/basics/android.md

```diff
@@ -46,19 +53,45 @@ skygear.signupWithEmail(email, password, new AuthResponseHandler() {
     }
 
     @Override
-    public void onAuthFail(String reason) {
-        Log.i("Skygear Signup", "onAuthFail: Fail with reason: " + reason);
+    public void onAuthFail(Error error) {
+        Log.i("Skygear Signup", "onAuthFail: Fail with reason: " + error.getMessage());
     }
 });
 ```
 
-## Logging in
+### Anonymous user
 
-The following shows how to log in a user with username and password.
+Skygear support users to sign up anonymously. However, they can only read data from the public database, but cannot perform operations such as saving data.
```

Comment:
@tensiuyan @Camerash I read the JS guide again and it is actually pretty clear about it. I think we might want to change this line to make it clearer: 

> Without being authenticated, a user can read data from the public database but cannot perform most of the other operations, including saving data into the database.

to this

> Without signing in, a client can only read data from the public database but cannot perform most of the other operations, including saving data into the database.

Just don't mix up about "authenticated" and "anonymous" user.